### PR TITLE
Update readme issue 40

### PR DESCRIPTION
**Pull request by adisos**

### CODE REVIEW

File: pkg/ibmvpc/nacl_analysis.go [hunk 1]
```diff
@@ -136,7 +134,7 @@ func (r *NACLRule) dumpRule() string {
 	return fmt.Sprintf("src: %s, dst: %s, conn: %s, action: %s", r.src.ToIPRanges(), r.dst.ToIPRanges(), r.connections.String(), r.action)
 }
 
-func (na *NACLAnalyzer) dumpNACLrules() string {
+func (na *NACLAnalyzer) DumpNACLrules() string {
```
Comment (zivnevo) [inline pkg/ibmvpc/nacl_analysis.go:137]: Why exporting all these dumping functions? Is it easier during debugging?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,52 @@
 # vpc-network-config-analyzer
-A tool for analyzing the network connectivity of a VPC as specified by various VPC resources.
+
+## About vpc-network-config-analyzer
+This repo contains a CLI for analyzing the network connectivity of a VPC as specified by various VPC resources.
```
Comment (zivnevo) [inline README.md:4]:
```suggestion
This repo contains packages and a CLI for analyzing the network connectivity of a VPC as specified by various VPC resources.
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,52 @@
...
+        whether to group together src/dst entries with identical connectivity
+  -output-file string
+        file path to store results
+  -vpc-config string
+        file path to input config
+
+```
+
+### Input config file
+The input config file should be the `json` file produced by the [`cloud-resource-collector`](https://github.com/np-guard/cloud-resource-collector)
```
Comment (zivnevo) [inline README.md:25]:
```suggestion
The input config file should be a `JSON` file produced by the [`cloud-resource-collector`](https://github.com/np-guard/cloud-resource-collector)
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,52 @@
...
+
+## Understanding the output
+
+### all_endpoints analysis type
+Each output line is of the form: `src => dst : connection` , where `src`,`dst` are either VPC endpoints (instance network interface) or external CIDR, and `connection` is the set of allowed protocols and their relevant connection attributes (e.g. source ports or destination ports for TCP/UDP).
```
Comment (zivnevo) [inline README.md:30]:
```suggestion
Each output line is of the form: `src => dst : connection` , where each of `src` and `dst` is either a VPC endpoint (instance network interface) or an external CIDR, and `connection` is the set of allowed protocols and their relevant connection attributes (e.g., allowed source ports and/or destination ports for TCP/UDP).
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,52 @@
...
+
+### all_subnets analysis type 
+Each output line is of the form: `src => dst : connection` , where `src`,`dst` are either VPC subnets or external CIDR, and `connection` is as explained above.
```
Comment (zivnevo) [inline README.md:33]:
```suggestion
Each output line is of the form: `src => dst : connection` , where each of `src` and `dst` is either a VPC subnet or an external CIDR, and `connection` is as explained for `all_endpoints`.
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,52 @@
...
+
+### single_subnet analysis type 
+The output consists of sections per each subnet (section header is the subnet CIDR block). Each section consists of two sub-sections: ingressConnectivity and egressConnectivity. These sections detail the allowed connectivity to/from the subnet, as configured by the subnet's NACL resource.
```
Comment (zivnevo) [inline README.md:36]:
```suggestion
The output consists of sections; one section per subnet (section header is the subnet's CIDR block). Each section consists of two sub-sections: `ingressConnectivity` and `egressConnectivity`. These sections detail the allowed connectivity to/from the subnet, as configured by the subnet's NACL resource.
```

**Review by zivnevo — CHANGES_REQUESTED** [review]
A few tiny things

File: pkg/ibmvpc/nacl_analysis.go (hunk 1 above)
Comment (adisos) [reply pkg/ibmvpc/nacl_analysis.go:137]: They are currently not used but I still want to keep them, and when exporting there are no warnings for those unused functions.

Comment (zivnevo) [reply pkg/ibmvpc/nacl_analysis.go:137]: I would prefer not to export them. Before you even notice, someone might consume this library and actually rely on the output of these functions, and then you have to maintain them forever :wink:

Either add a lint exception or add something like
```golang
var _ = (*ResourcesContainer).printDetails // avoiding "unused" warning
```

Comment (adisos) [reply pkg/ibmvpc/nacl_analysis.go]: ok, changed accordingly

**Review by zivnevo — APPROVED** [review]

**Pull request merged**